### PR TITLE
Feedback

### LINKED ISSUE #73: Connection failure notification
# User Story: 
- As a medical professional I want to be notified in the event of not being able to connect to the SmartCar, so that I will be aware of the car’s availability.

# Acceptance criteria:
- When connection between the Android application and the SmartCar cannot be established, a notification and a voice confirmation as well as well as a vibration shall be displayed within 10 seconds.

### PR DESCRIPTION
**Pull request by SandraSmolerEisenberg**
# Related issue
- Helps #72, closes #73, Helps #61 

# Proposed changes
- Add feedback for connection failure.
- Add feedback for connection success.
- Add feedback for obstacle detection.
- Add tests for the corresponding features

# Additional information
- Feedback for obstacle detection is implemented in the application only. The SmartCar sketch needs to be updated accordingly. 
- The Codecov report tool is still under set-up and the team has not decided if there are to use it or not

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ env:
     # Set the minimum Android API to support
     # Note: Not all of them work well with Travis
     - ANDROID_TARGET=android-21
+    # Codcov token
+    - CODECOV_TOKEN="b2a0428d-3c42-4c5e-b7c6-1563a33e76c6"
```
Comment (platisd) [inline .travis.yml]: Should you be committing this?

Comment (krasen86) [reply .travis.yml]: I tried to run it without the token in the previous build  but it didn't work

Comment (krasen86) [reply .travis.yml]: The documentation says it is not needed for public repositories, We are still setting it up

Comment (platisd) [reply .travis.yml]: You should set it as a (secret) environment variable in Travis, don't expose it. Now it's compromised so I'd suggest generating a new one.

Comment (krasen86) [reply .travis.yml]: Yes sure That's not a problem, we are not sure if we want to use the service at all, so if we manage to set it up we'll update it accordingly.

Comment (krasen86) [reply .travis.yml]: Apparently I'm not able to trigger a new build with the new Travis config, Do you know how to trigger one or do I need to close and reopen the pull request?

**Comment by krasen86** [comment]
I manage to resolve the issue. For some reason, I was not able to trigger a build on the latest commit, and Travis was running the builds on the old yml version which didn't allow us to synchronize with codecov.io,  It works fine now after committing the update to the sprint branch and a new token has been generated.

File: .travis.yml (hunk 1 above)
Comment (krasen86) [reply .travis.yml]: A new token has been generated btw :)

Comment (platisd) [reply .travis.yml]: I've seen codecov getting stuck or in a bad shape sometimes.
You can trigger a new build if you either push something new, create a new pull request or manually run the codecov command you have in your Travis script **while** having set the appropriate environment variable to your API token.

**Review by chrytsa — CHANGES_REQUESTED** [review]
Based on the traceability used for this pull request, I verify the following:

- for issue #72: `ResponseHandler` class is inline using `handleObstacleDetection()` method
- for issue #73: `handleFailure()` currently implements logic to vibrate on unsuccessful initial connection, without any corresponding acceptance criteria on the respective User Stories. 
Furthermore, as an extra note, the name of the method doesn't explain what kind of failure it handles without inspecting the code. So, I think it would be better either to use a self-explanatory name, for instance `handleConnectionRequest`, or provide some comments explaining the logic of the method as you did for handleSuccess.

Therefore, the acceptance criteria probably would need to be revised before approving the request.

**Pull request merged**